src/app/zones/sources.ts

```diff
@@ -38,30 +38,22 @@ export const sources = (source: Source, api: KumaApi) => {
       return api.getZoneOverview({ name })
     },
     '/zone-cps/online/:name': (params: DetailParams) => {
-      // this source retries until we have a 200 and the zone found is online
-      // any non-404 errors will error as usual
+      const ZoneOfflineError = class extends Error {}
       const { name } = params
       return source(async () => {
         const res = await api.getZoneOverview({ name })
+        // The presence of a `ZoneOverview.zoneInsight` object's subscriptions
+        // with a connect time and without a disconnect time indicate a Zone to
+        // be connected and online.
         if (getZoneControlPlaneStatus(res) === 'online') {
           return res
         } else {
-          const e = new ApiError({
-            status: 404,
-            title: `The ${res.name} Zone is offline`,
-          })
-          throw e
+          throw new ZoneOfflineError()
         }
       }, {
         retry: (e) => {
-          const hasStatus = <T extends {status: number}>(e: unknown): e is T => typeof (e as T).status !== 'undefined'
-          if (hasStatus(e)) {
-            const status = e.status.toString()
-            switch (true) {
-              case status === '404':
-                // wait 2 seconds and try again
-                return new Promise((resolve) => setTimeout(resolve, 2000))
-            }
+          if (e instanceof ZoneOfflineError) {
+            return new Promise((resolve) => setTimeout(resolve, 2000))
```

Comment:
The actual fix is here. We no longer consider a 404 a signal to retry, only a 'zone offline' would trigger a retry.